Fix problems with wildcards inside web scan header configuration `onlyForUrls`

## Situation
With #2174 we introduced a wildcard `<*>` for filtering urls by `onlyForUrls`.

We wrote an integration test, but in real server deployment the communication between SecHub client and SecHub server failed when creating a new SecHub job.

The failure message contained information that it is not a valid url - but the text did not contain `<*>` but instead `lt;*` - means html escaped

## Wanted
Find the problem and fix it.

## Solution
- Enhance / or write new integration test to reprocude the problem (and have a failing build)
- Fix the problem

The integration test `PDSWebScanJobScenario12IntTest` does now use SecHub client and reproduces the situation: It does currently fail with:

```java
java.lang.AssertionError: Not exit code 0 but:5 , last output line was:body={"status":400,"error":"Bad Request","message":"Validation failed for object='secHubConfiguration'. Error count: 2","details":["The URL for a HTTP header is not a valid URL! OnlyForUrls defined URL: https://demo.example.org/myapp/upload/&lt;*>/ is not a valid URL.","The URL for a HTTP header is not a valid URL! OnlyForUrls defined URL: https://demo.example.org/myapp/&lt;*>/special/ is not a valid URL."],"timeStamp":"Thu Aug 03 15:44:13 CEST 2023"}
	at org.junit.Assert.fail(Assert.java:89)
	at com.mercedesbenz.sechub.integrationtest.api.WithSecHubClient.startAsynchronScanFor(WithSecHubClient.java:268)
	at com.mercedesbenz.sechub.integrationtest.api.WithSecHubClient.startAsynchronScanFor(WithSecHubClient.java:293)
	at com.mercedesbenz.sechub.integrationtest.api.WithSecHubClient.startAsynchronScanFor(WithSecHubClient.java:277)
	at com.mercedesbenz.sechub.integrationtest.scenario12.PDSWebScanJobScenario12IntTest.pds_web_scan_has_expected_info_finding_with_given_target_url_and_product2_level_information_and_sechub_web_config_parts(PDSWebScanJobScenario12IntTest.java:56)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.mercedesbenz.sechub.integrationtest.api.IntegrationTestSetup$IntegrationTestStatement.evaluate(IntegrationTestSetup.java:229)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:833)


```

--- a/sechub-cli/src/mercedes-benz.com/sechub/cli/resthelper.go
+++ b/sechub-cli/src/mercedes-benz.com/sechub/cli/resthelper.go
@@ -50,7 +50,7 @@ func handleHTTPRequestAndResponse(context *Context, request *http.Request, print
 		sechubUtil.LogDebug(true, fmt.Sprintf("HTTP %s %s\n  HTTP Headers: %s\n  Content length of HTTP request: %d bytes",
 			request.Method, request.URL, request.Header, request.ContentLength))
 		if printContentOfRequest {
-			sechubUtil.LogDebug(true, fmt.Sprintf("Content of HTTP request: %q", context.inputForContentProcessing))
+			sechubUtil.LogDebug(true, fmt.Sprintf("Content of HTTP request: %q", request.Body))
 		}
 	} else {
 		sechubUtil.LogDebug(context.config.debug, fmt.Sprintf("HTTP %s %s", request.Method, request.URL))

--- a/sechub-cli/src/mercedes-benz.com/sechub/cli/sechubconfig.go
+++ b/sechub-cli/src/mercedes-benz.com/sechub/cli/sechubconfig.go
@@ -6,11 +6,11 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
-	"html/template"
 	"io"
 	"io/ioutil"
 	"os"
 	"strings"
+	"text/template"
 
 	sechubUtil "mercedes-benz.com/sechub/util"
 )

--- a/sechub-integrationtest/src/main/java/com/mercedesbenz/sechub/integrationtest/api/WithSecHubClient.java
+++ b/sechub-integrationtest/src/main/java/com/mercedesbenz/sechub/integrationtest/api/WithSecHubClient.java
@@ -15,6 +15,9 @@ import java.util.UUID;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.mercedesbenz.sechub.commons.TextFileWriter;
+import com.mercedesbenz.sechub.commons.model.JSONConverter;
+import com.mercedesbenz.sechub.commons.model.SecHubScanConfiguration;
 import com.mercedesbenz.sechub.integrationtest.api.AsUser.ProjectFalsePositivesDefinition;
 import com.mercedesbenz.sechub.integrationtest.internal.IntegrationTestExampleConstants;
 import com.mercedesbenz.sechub.integrationtest.internal.IntegrationTestFileSupport;
@@ -253,6 +256,11 @@ public class WithSecHubClient {
     public AssertAsyncResult startAsynchronScanFor(TestProject project, IntegrationTestJSONLocation location, Map<String, String> environmentVariables,
             ApiTokenStrategy apiTokenStrategy) {
         File sechubConfigFile = IntegrationTestFileSupport.getTestfileSupport().createFileFromResourcePath(location.getPath());
+        return startAsynchronScanFor(project, environmentVariables, apiTokenStrategy, sechubConfigFile);
+    }
+
+    public AssertAsyncResult startAsynchronScanFor(TestProject project, Map<String, String> environmentVariables, ApiTokenStrategy apiTokenStrategy,
+            File sechubConfigFile) {
         SecHubClientExecutor executor = createExecutor();
         List<String> list = buildEnvironmentAndBehaviourCommands(project);
         ExecutionResult result = doExecute(ClientAction.START_ASYNC, apiTokenStrategy, sechubConfigFile, executor, list, environmentVariables);
@@ -263,6 +271,29 @@ public class WithSecHubClient {
         asynchResult.jobUUID = UUID.fromString(result.getLastOutputLine());
         asynchResult.configFile = sechubConfigFile;
         return asynchResult;
+    }
+
+    public AssertAsyncResult startAsynchronScanFor(TestProject project, SecHubScanConfiguration configuration) {
+        return startAsynchronScanFor(project, configuration, null, ApiTokenStrategy.VISIBLE_AS_ARGUMENT);
+    }
+
+    public AssertAsyncResult startAsynchronScanFor(TestProject project, SecHubScanConfiguration configuration, Map<String, String> environmentVariables,
+            ApiTokenStrategy apiTokenStrategy) {
+        /* ensure project id is inside the configuration */
+        configuration.setProjectId(project.getProjectId());
+
+        String configAsJson = JSONConverter.get().toJSON(configuration, true);
+        try {
+            Path tempFile = TestUtil.createTempFileInBuildFolder("sechub-config", ".json");
+            File file = tempFile.toFile();
+
+            TextFileWriter writer = new TextFileWriter();
+            writer.save(file, configAsJson, true);
+
+            return startAsynchronScanFor(project, environmentVariables, apiTokenStrategy, file);
+        } catch (IOException e) {
+            throw new IllegalStateException("Was not able to create test temp file!", e);
+        }
     }
 
     /**
@@ -450,4 +481,5 @@ public class WithSecHubClient {
 
         return asUser.create(project, json);
     }
+
 }

--- a/sechub-integrationtest/src/test/java/com/mercedesbenz/sechub/integrationtest/scenario12/PDSWebScanJobScenario12IntTest.java
+++ b/sechub-integrationtest/src/test/java/com/mercedesbenz/sechub/integrationtest/scenario12/PDSWebScanJobScenario12IntTest.java
@@ -51,12 +51,9 @@ public class PDSWebScanJobScenario12IntTest {
         TestProject project = PROJECT_1;
         String targetURL = configuration.getWebScan().get().getUrl().toString();
         as(SUPER_ADMIN).updateWhiteListForProject(project, Arrays.asList(targetURL));
-        UUID jobUUID = as(USER_1).createJobAndReturnJobUUID(project, configuration);
 
         /* execute */
-        as(USER_1).
-            approveJob(project, jobUUID);
-
+        UUID jobUUID = as(USER_1).withSecHubClient().startAsynchronScanFor(project, configuration).getJobUUID();
         waitForJobDone(project, jobUUID, 30, true);
 
         /* test */

